### ISSUE TITLE
first-commit.com is not doing anything

### ISSUE THREAD

**Issue by balupton** [wong2/first-commit#15]
perhaps deploy to now.sh and provide the now URL? would save the cost of the domain name

**Comment by kevinkassimo** [comment]
Also found the site down days ago. Thus I wrote a version that also adds sequential commit browsing and is now available at http://buhtig.com (reverse of Github lol)

**Comment by wong2** [comment]
yeah, this is due to Github's changes on their front-end code, and I haven't found a way to make first-commit work again..

**Comment by wong2** [comment]
@kevinkassimo I can't open your site T T

**Comment by kevinkassimo** [comment]
@wong2 It seems my node server has just stopped running. I am restarting it now

**Comment by kevinkassimo** [comment]
@wong2 Back up again. It seems my server killed the nohup process for some weird reason. Restarted with forever command instead. For Buhtig.com, I added usage of Github API. （说起来你介意我在repo里提到你的id吗（因为参考了你一部分的idea。Github API自带的commit数有误所以commit还是用了headless browser jquery html parsing）Orz以及自己其实才刚刚开始学node。。。）

**Comment by wong2** [comment]
@kevinkassimo 没问题。所以你现在是怎样获取 fist commit 的？自从 Github 改成 https://github.com/git/git/commits/master?after=b14f27f91770e0f99f64135348977a0ce1c7993a+34 这种翻页模式后我还没找到方法

**Comment by kevinkassimo** [comment]
@wong2 api.github.com，它会按页提供commit info的json，所以先parse对应repo的页面获得commit总数（Github API自带的commit数这方面有偏差），然后计算应该向api.github.com索取第几页的信息（所以不仅可以获取first commit，而实际上可以获得任何一个sequentially labeled commit）https://developer.github.com/v3/#pagination。不过Github API有次数上限，所以网站暂时每小时最多query 1000次来着（虽说一般够用了）。。。可能之后想试试看改成Github Login的话就可以依赖每个user独立的query limit，可能会好很多。。。

**Comment by wong2** [comment]
@kevinkassimo thanks! 我试试

**Comment by chocolateboy** [comment]
There are a few snippets for doing this e.g.:

* https://gist.github.com/JamesWP/bc3622dd3d94bf6cbeb5c83b7f183b55
* https://gist.github.com/simonewebdesign/a70f6c89ffd71e6ba4f7dcf7cc74ccf8
* https://github.com/FarhadG/init/issues/6
* https://github.com/FarhadG/init

**Comment by kevinkassimo** [comment]
Has been a while. I have again revamped my replacement site doing similar job, [buhtig.com](https://buhtig.com) with React & Go, and changed commit fetching logic again, now it's much faster with caching! Hope you guys who need it loves the new site!

**Comment by kevinkassimo** [comment]
@chocolateboy You should check out my site too, At [buhtig.com](https://buhtig.com) (domain is the reverse of Github LOL)

**Comment by chocolateboy** [comment]
@kevinkassimo Thanks, already spotted! (I mention it [here](https://greasyfork.org/en/scripts/38557-github-first-commit) :-)
